sympy/series/limitseq.py

```diff
@@ -165,35 +202,16 @@ def limit_seq(expr, n=None, trials=5):
     elif n not in expr.free_symbols:
         return expr
 
-    for i in range(trials):
-        if not expr.has(Sum):
-            result = _limit_inf(expr, n)
-            if result is not None:
-                return result
-
-        num, den = expr.as_numer_denom()
-        if not den.has(n) or not num.has(n):
-            result = _limit_inf(expr.doit(), n)
-            if result is not None:
-                return result
-            return None
-
-        num, den = (difference_delta(t.expand(), n) for t in [num, den])
-        expr = (num / den).gammasimp()
-
-        if not expr.has(Sum):
-            result = _limit_inf(expr, n)
-            if result is not None:
-                return result
-
-        num, den = expr.as_numer_denom()
-
-        num = dominant(num, n)
-        if num is None:
-            return None
-
-        den = dominant(den, n)
-        if den is None:
-            return None
-
-        expr = (num / den).gammasimp()
+    n_ = Dummy("n", integer=True, positive=True)
+
+    # If there is a negative term raised to a power involving n, consider
+    # even and odd n separately.
+    powers = (p.as_base_exp() for p in expr.atoms(Pow))
+    if any(b.is_negative and e.has(n) for b, e in powers):
+        L1 = _limit_seq(expr.xreplace({n: 2*n_}), n_, trials)
+        if L1 is not None:
+            L2 = _limit_seq(expr.xreplace({n: 2*n_ + 1}), n_, trials)
+            if L1 == L2:
+                return L1
+    else:
+        return _limit_seq(expr.xreplace({n: n_}), n_, trials)
```

Comment:
Why this `xreplace`?

Reply:
This is what I referred to in the test as "the index is now declared to be positive and integer, in case the user forgot to do so". For example, in master branch
```
var('n')
limit_seq(cos(2*pi*n), n)
```
returns AccumBounds(-1, 1). With this patch it returns 1.

Reply:
That is a good point, overlooked in the original implementation.